### PR TITLE
Add scripts for building ObjWriter

### PR DESCRIPTION
**Pull request by AntonLapounov**
Add scripts to build ObjWriter if `NativeAot.ObjWriter` subset is specified. Also add `FK_PCRel_4` handling in the `AArch64WinCOFFObjectWriter::getRelocType` function to fix an ObjWriter error when ILC targets ARM64.

### CODE REVIEW

File: src/coreclr/Tools/aot/ObjWriter/build.cmd [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+:: Copy ObjWriter files
+robocopy /mir "%~dp0\" tools\ObjWriter
+if %ErrorLevel% geq 8 exit /b %ErrorLevel%
+
+:: Configure and build
+if not exist build mkdir build
+cd build || exit /b 1
+
+cmake ../ -DCMAKE_BUILD_TYPE=Release -DLLVM_OPTIMIZED_TABLEGEN=1 -DHAVE_POSIX_SPAWN=0 -DLLVM_ENABLE_PIC=1 -DLLVM_BUILD_TESTS=0 -DLLVM_ENABLE_DOXYGEN=0 -DLLVM_INCLUDE_DOCS=0 -DLLVM_INCLUDE_TESTS=0 -DLLVM_TARGETS_TO_BUILD="ARM;X86;AArch64" -DCMAKE_INSTALL_PREFIX=install -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=WebAssembly || exit /b 1
```
Comment (jkotas) [inline src/coreclr/Tools/aot/ObjWriter/build.cmd:22]: We can omit WebAssembly here

File: src/coreclr/Tools/aot/ObjWriter/build.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Add ObjWriter files
+rsync -av --delete "$ScriptPath/" tools/ObjWriter || exit 1
+
+# Configure and build
+[ -d build ] || mkdir build
+cd build || exit 1
+
+cmake ../ -DCMAKE_BUILD_TYPE=Release -DLLVM_OPTIMIZED_TABLEGEN=1 -DHAVE_POSIX_SPAWN=0 -DLLVM_ENABLE_PIC=1 -DLLVM_BUILD_TESTS=0 -DLLVM_ENABLE_DOXYGEN=0 -DLLVM_INCLUDE_DOCS=0 -DLLVM_INCLUDE_TESTS=0 -DLLVM_TARGETS_TO_BUILD="ARM;X86;AArch64" -DCMAKE_INSTALL_PREFIX=install -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=WebAssembly || exit 1
+cmake --build . -j 10 --config Release --target install || exit 1
```
Comment (jkotas) [inline src/coreclr/Tools/aot/ObjWriter/build.sh:23]: Does the build place the binary in the right directory to be included in the package; or is that part left for next PR?

**Review by jkotas — APPROVED** [review]
Thank you!

File: src/coreclr/Tools/aot/ObjWriter/build.sh (hunk 2 above)
Comment (AntonLapounov) [reply src/coreclr/Tools/aot/ObjWriter/build.sh:23]: Not yet.

File: src/coreclr/Tools/aot/ObjWriter/build.cmd (hunk 1 above)
Comment (AntonLapounov) [reply src/coreclr/Tools/aot/ObjWriter/build.cmd]: Removed WebAssembly.

**Pull request merged**